### PR TITLE
fix console tab error

### PR DESCRIPTION
**Pull request by sarthak212**
#3972

### CODE REVIEW

File: mitmproxy/tools/console/tabs.py [hunk 1]
```diff
@@ -50,7 +50,7 @@ def show(self):
         headers = []
         for i in range(len(self.tabs)):
             txt = self.tabs[i][0]()
-            if i == self.tab_offset:
+            if i == self.tab_offset or (len(self.tabs) <= self.tab_offset and i == 0):
```
Comment (mhils) [inline mitmproxy/tools/console/tabs.py:53]: Style tip for this type of code: At the moment it's really hard to parse what is actually happening here. You can alleviate this by introducing a variable with a speaking name:
```python
is_focused = (
    i == self.tab_offset
    or (i == 0 and self.tab_offset >= len(self.tabs)):
)
if is_focused:
    # ...
```

I like that you don't modify `.tab_offset` if we have less tabs: That would be annoying when tabbing HTTP flow -> TCP flow -> HTTP flow as it would reset the offset. We could however simplify the implementation a bit and use `self.tab_offset % len(self.tabs)` everywhere.

File: mitmproxy/tools/console/tabs.py [hunk 2]
```diff
@@ -70,7 +70,7 @@ def show(self):
                 )
         headers = urwid.Columns(headers, dividechars=1)
         self._w = urwid.Frame(
-            body = self.tabs[self.tab_offset][1](),
+            body = self.tabs[self.tab_offset if len(self.tabs) < self.tab_offset else 0][1](),
```
Comment (mhils) [inline mitmproxy/tools/console/tabs.py:73]: `<` or `<=`? (or `% len(self.tabs)` as mentioned above :))

File: mitmproxy/tools/console/tabs.py (hunk 1 above)
Comment (sarthak212) [reply mitmproxy/tools/console/tabs.py]: Thanks for the tip

**Comment by Kriechi** [comment]
@mhils good to go?

**Comment by mhils** [comment]
Thanks! 🍰
